docs/reference/filebeat/delete-file-guide.md

```diff
@@ -0,0 +1,134 @@
+# Removing files after ingestion
+
+::::{warning}
+Enabling this feature will remove files, which could lead to data loss.
+::::
+
+The Filestream input can remove files after they have been fully
+ingested, three requirements need to be met before the Filestream
+input can remove a file:
+1. The reader for the file needs to be closed;
+2. All events from the file have been published (acknowledged by the
+   output);
```

Comment:
suggestion only, not a blocking comment.

This is correct, but the terms are pretty specific to how things are named in our implementation.  I'm thinking it might be useful to use more generic terms that closer to how a user "sees" the flow.

1. conditions have been met for filestream to close the file.  This is controlled by the `close.*` options.
2. events from the file have been received by the configured output without error.  (example elasticsearch output has indexed event or logstash has written event to persistent queue)
3. the `delete.grace_period` has expired.  This means nothing happened for filestream to re-open the file during that time.  The most common thing to cause filestream to re-open the file is adding data to it.